clang/test/CodeGenSYCL/intel-fpga-loops.cpp

```diff
@@ -150,6 +158,44 @@ void loop_count_control() {
       a[i] = 0;
 }
 
+// Add CodeGen tests for Loop attribute: [[intel::fpga_pipeline()]].
+template <int A>
+void fpga_pipeline() {
+  int a[10];
+  // CHECK: ![[MD_FP]] = distinct !{![[MD_FP]], ![[MP]], ![[MD_fpga_pipeline:[0-9]+]]}
+  // CHECK-NEXT: ![[MD_fpga_pipeline]] = !{!"llvm.loop.intel.pipelining.enable", i32 1}
+  [[intel::fpga_pipeline(A)]] for (int i = 0; i != 10; ++i)
+    a[i] = 0;
+
+  // CHECK: ![[MD_FP_1]] = distinct !{![[MD_FP_1]], ![[MP]], ![[MD_fpga_pipeline]]}
```

Comment:
Can you add the check for metadata lvm.loop.intel.pipelining.enable in each of these cases as well?

Reply:
Thanks @elizabethandrews for reviews. I tried but it looks like it is not possible to add metadata for each tests. This is same for other attributes as well. They do not check metadata for each test cases.

```
!56 = !{!"llvm.loop.intel.pipelining.enable", i32 1}
!57 = distinct !{!57, !12, !56}
!58 = distinct !{!58, !12, !56}
!59 = distinct !{!59, !12, !13}
!60 = distinct !{!60, !12, !56}
!61 = distinct !{!61, !12, !56}
!62 = distinct !{!62, !56}
!63 = distinct !{!63, !12, !56}
```